app/main/viewmanager.ts

```diff
@@ -41,12 +41,9 @@ class ViewManager {
 		});
 
 		ipcMain.on('call-view-function', (e: Event, name: string, ...params: any[]) => {
-			const fn = this.views[this.selectedIndex][name as keyof View];
-			if (typeof fn === 'function') {
-				// Type checking requires spread elements to match up with a rest parameter.
-				// So, using a workaround here.
-				(fn as any)(...params);
-			}
+			// Type checking requires spread elements to match up with a rest parameter.
+			// So, using a workaround here.
+			(this.views[this.selectedIndex] as any)[name as keyof View](...params);
```

Comment:
I wonder if `(this.views[this.selectedIndex] as View)` work here?

Reply:
No, can't. Getting this error-
```
Cannot invoke an expression whose type lacks a call signature. Type 'string | number | boolean | ViewProps | WebContents | (() => boolean) | ((event: string, listener: Function) => View) | ((event: string, listener: Function) => View) | ((event: string, listener: Function) => View) | ... 27 more ... | (() => string[])' has no compatible call signatures.ts(2349)
```

Reply:
OK, so I think I know why this throws error. The correct way to write this would be:
```typescript
type ViewCallable = (...params: any[]) -> any;
(this.views[this.selectedIndex][name as keyof View] as ViewCallable)(...params);
```

Reply:
Since we do this in couple of places I do split this off into it's own function and also check if it a function before calling it too. So something like:
```typescript
callViewFunction(index: number, name: keyof View, ...params; any[]) {
    const viewFunction = this.views[index][name];
    if (typeof viewFunction === 'function') {
        type ViewCallable = (...args: any[]) -> any;
        (viewFunction as ViewCallable)(...params);
    } else {
         // error reporting here
    }
}
```

Reply:
I have made a separate function in the last commit, but couldn't make this work. Get webcontents not defined error when i use this. See if you can make it work.